src/patterns.jl

```diff
@@ -0,0 +1,63 @@
+
+"""
+    Pattern
+
+A pattern is just a wrapper around a `Tuple{Vararg{Symbol}}` which enables searching and
+filtering for matching components and dimension paths in a [`Dataset`](@ref).
+Special symbols `:_` and `:__` are used as wildcards, similar to `*` and `**` in glob
+pattern matching.
+"""
+struct Pattern
+    segments::Tuple{Vararg{Symbol}}
+end
+
+Pattern(segments::Symbol...) = Pattern(segments)
+Base.convert(::Type{Pattern}, x::Tuple{Vararg{Symbol}}) = Pattern(x)
+
+# Primary functionality is to support identify tuples that match the constraint
```

Comment:
Do you mean "support identity tuples" or "is to identify tuples"

Reply:
The later, thanks.